### PR TITLE
Cleanup disk cache io device

### PR DESCRIPTION
**Pull request by Atlante45**
Fix memory/file descriptors leak

### CODE REVIEW

**Comment by Atlante45** [comment]
https://app.asana.com/0/26551373415487/113900691717503

**Comment by hifi-gustavo** [comment]
Build console output for commit: c1d9f8fcfceca2fb2a9e1cbe5c56b3ab1a4addf0 available at: https://gist.github.com/0aa8823157c07dbcd72ce1c0889fadcb<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7700.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7700.dmg

File: libraries/networking/src/AssetUtils.cpp [hunk 1]
```diff
@@ -29,7 +31,7 @@ QByteArray hashData(const QByteArray& data) {
 
 QByteArray loadFromCache(const QUrl& url) {
     if (auto cache = NetworkAccessManager::getInstance().cache()) {
-        if (auto ioDevice = cache->data(url)) {
+        if (auto ioDevice = std::unique_ptr<QIODevice>(cache->data(url))) {
```
Comment (sethalves) [inline libraries/networking/src/AssetUtils.cpp]: clement would you mind putting a comment here that says something like: use unique_ptr to delete what cache->data(url) returns?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8e0ade64251ac32f3930fac2a67d838caba776d4 available at: https://gist.github.com/0aa8823157c07dbcd72ce1c0889fadcb<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7700.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7700.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b5667b73dc968736c3831f2afb706a166cddcf7d available at: https://gist.github.com/0aa8823157c07dbcd72ce1c0889fadcb<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7700.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7700.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c9fa91936a96d3999186e2850599034a24b49814 available at: https://gist.github.com/0aa8823157c07dbcd72ce1c0889fadcb<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7700.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7700.dmg

**Comment by sethalves** [comment]
I've built and tested this.

**Pull request merged**